File: DAA.Web.Intranet/client/src/models/responseResult.ts
import { IResponseResult } from "@/interfaces/responseResult";

export class ResponseResult {
  constructor(obj: IResponseResult) {
    Object.assign(this, obj);
    this.success = obj.success;
    this.code = obj.code;
    this.showMessage = obj.showMessage;
    //this.data = obj.data;
  }
  success: boolean;
  code: number;
  message?: string;
  showMessage: boolean;
  data?: unknown;
}

File: DAA.Shared/ResponseResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DAA.Shared
{
    public class ResponseResult
    {
        public bool Success { get; set; }
        public int Code { get; set; }
        public string? Message { get; set; }
        public bool ShowMessage { get; set; } = false;
    }

    public class ResponseResult<T>
    {
        public bool Success { get; set; }
        public int Code { get; set; }
        public string? Message { get; set; }
        public bool ShowMessage { get; set; } = false;
        public T? Data { get; set; }
    }
}
